### PR TITLE
BaSP-TG: superadmins methods (create, delete and filter)

### PR DESCRIPTION
**Pull request by DinoStampella**
1) Code the logic to create superadmins
2) Code the logic to delet superadmins
3) Code the logic to get superadmins by filters

### CODE REVIEW

File: src/resources/super-admins.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      } else {
+        res.send('SuperAdmin deleted successfully');
+      }
+    });
+  } else {
+    res.send('Non-existing SuperAdmin');
+  }
+});
+
+// GET BY FILTERS
```
Comment (Licamus) [inline src/resources/super-admins.js:43]: It is possible to filter with more than 1 parameter at the same time?

Comment (DinoStampella) [reply src/resources/super-admins.js:43]: In this case, you can filter one parameter at a time.

File: src/resources/super-admins.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+const fs = require('fs');
+const superAdmins = require('../data/super-admins.json');
+
+const router = express.Router();
+
+// CREATE SUPER-ADMIN
+router.post('/add', (req, res) => {
+  const newSuperAdmin = req.body;
+  if (Object.keys(newSuperAdmin).length !== 5) {
```
Comment (ignacioaristo) [inline src/resources/super-admins.js:10]: With this validation, we can enter any key, no matter the name.

File: src/resources/super-admins.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    res.send('Please, complete all fields');
+    return;
+  }
+  const newID = Number(superAdmins[superAdmins.length - 1].id) + 1;
```
Comment (ignacioaristo) [inline src/resources/super-admins.js:14]: Good one, the next week we are not going to need this, the id will be generated automatically.

File: src/resources/super-admins.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  newSuperAdmin.id = newID.toString();
+  superAdmins.push(newSuperAdmin);
+  fs.writeFile('src/data/super-admins.json', JSON.stringify(superAdmins), (err) => {
+    if (err) {
+      res.send('Can not save new SuperAdmin');
+    } else {
+      res.send('SuperAdmin created successfully');
```
Comment (ignacioaristo) [inline src/resources/super-admins.js:21]: We can show the new data, for e.g:
`      res.send('SuperAdmin created successfully: ${newSuperAdmin}');`

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/resources/super-admins.js (hunk 3 above)
Comment (DinoStampella) [reply src/resources/super-admins.js:14]: Great! So do I remove this validation?

Comment (ignacioaristo) [reply src/resources/super-admins.js:14]: I would say yes, just in case.

File: src/resources/super-admins.js (hunk 2 above)
Comment (DinoStampella) [reply src/resources/super-admins.js:10]: Validation added

File: src/resources/super-admins.js (hunk 3 above)
Comment (DinoStampella) [reply src/resources/super-admins.js]: newId removed

File: src/resources/super-admins.js (hunk 4 above)
Comment (DinoStampella) [reply src/resources/super-admins.js]: Message updated

**Review by baudraccoverona — APPROVED** [review]
Good to go ✨

**Pull request merged**